packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -1812,6 +1831,74 @@ export class SeedlessOnboardingController<EncryptionKey> extends BaseController<
     });
   }
 
+  /**
+   * Revoke all pending refresh tokens.
+   *
+   * This method is to be called after user is authenticated.
+   *
+   * @returns A Promise that resolves to void.
+   */
+  async revokePendingRefreshTokens() {
+    return await this.#withControllerLock(async () => {
+      this.#assertIsAuthenticatedUser(this.state);
+      const { pendingToBeRevokedTokens } = this.state;
+      if (!pendingToBeRevokedTokens || pendingToBeRevokedTokens.length === 0) {
+        return;
+      }
+
+      // revoke all pending refresh tokens in parallel
+      const promises = pendingToBeRevokedTokens.map(({ revokeToken }) => {
+        const revokePromise = async (): Promise<string | null> => {
+          try {
+            await this.#revokeRefreshToken({
+              connection: this.state.authConnection as AuthConnection,
+              revokeToken,
+            });
+            return revokeToken;
+          } catch (error) {
+            log('Error revoking refresh token', error);
+            return null;
+          }
+        };
+        return revokePromise();
+      });
+      const result = await Promise.all(promises); // no need to do Promise.allSettled because the promise already handle try catch
+      // filter out the null values
+      const revokedTokens = result.filter((token) => token !== null);
+      if (revokedTokens.length > 0) {
+        // update the state to remove the revoked tokens once all concurrent token revoke finish
+        this.update((state) => {
+          state.pendingToBeRevokedTokens =
```

Comment:
what if `addRefreshTokenToRevokeList` is called when this state update is in progress.